# Be a little more paranoid about dereferencing pointers, because they may be NULL in the
# case of an empty document
#
# Copyright (C) 2006 - 2007 Nokia Corporation.
# This file is distributed under the terms of GNU LGPL license, either version 2
# of the License, or (at your option) any later version.
#
diff -ru gtkhtml-3.13.91/src/htmlengine-search.c gtkhtml-3.13.91.mod4/src/htmlengine-search.c
--- gtkhtml-3.13.91/src/htmlengine-search.c	2007-01-03 17:19:57.000000000 +0200
+++ gtkhtml-3.13.91.mod4/src/htmlengine-search.c	2007-02-15 14:14:16.000000000 +0200
@@ -141,6 +141,7 @@
 {
 	HTMLSearch *info;
 	HTMLObject *p;
+	HTMLObject *o;
 
 	if (e->search_info) {
 		html_search_destroy (e->search_info);
@@ -148,8 +149,9 @@
 
 	info = e->search_info = html_search_new (e, text, case_sensitive, forward, regular);
 
-	p = HTML_OBJECT (e->search_info->stack->data)->parent;
-	if (html_object_search (p ? p : e->clue, info)) {
+	p = e->search_info->stack ? HTML_OBJECT (e->search_info->stack->data)->parent : NULL;
+	o = p ? p : e->clue;
+	if (o && html_object_search (o, info)) {
 		display_search_results (info);
 		return TRUE;
 	} else
@@ -181,7 +183,7 @@
 			retval = html_object_search (HTML_OBJECT (info->stack->data), info);
 		else {
 			html_search_push (info, e->clue);
-			retval = html_object_search (e->clue, info);
+			retval = e->clue ? html_object_search (e->clue, info) : FALSE ;
 		}
 		if (retval)
 			display_search_results (info);
diff -ru gtkhtml-3.13.91/src/htmlsearch.c gtkhtml-3.13.91.mod4/src/htmlsearch.c
--- gtkhtml-3.13.91/src/htmlsearch.c	2007-02-09 21:54:58.000000000 +0200
+++ gtkhtml-3.13.91.mod4/src/htmlsearch.c	2007-02-15 14:14:52.000000000 +0200
@@ -67,11 +67,13 @@
 		for (o = e->cursor->object; o; o = o->parent)
 			html_search_push (ns, o);
 		ns->stack = g_slist_reverse (ns->stack);
-		ns->found = g_list_append (ns->found, e->cursor->object);
+		if (e->cursor->object)
+  		ns->found = g_list_append (ns->found, e->cursor->object);
 	} else {
 		ns->stack     = NULL;
 		ns->start_pos = 0;
-		html_search_push (ns, e->clue);
+		if (e->clue)
+			html_search_push (ns, e->clue);
 	}
 
 	/* translate table
